include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -407,7 +390,7 @@ template <class DataFacadeT, class Derived> class BasicRoutingInterface
                              : phantom_node_pair.target_phantom.forward_travel_mode});
         }
 
-        if (is_local_path && unpacked_path.size() > 0)
+        if (unpacked_path.size() > 0)
```

Comment:
this now fixes the very first segment in all cases, not only local cases (see above removal)